inginious/frontend/plugins/problem_bank/pages/api/manage_banks_courses_api.py

```diff
@@ -0,0 +1,49 @@
+import web
+
+from pymongo.errors import DuplicateKeyError
+from inginious.frontend.plugins.utils.admin_api import AdminApi
+from inginious.frontend.plugins.utils import get_mandatory_parameter
+
+
+class ManageBanksCoursesApi(AdminApi):
+    def get_course_id(self):
+        parameters = web.input()
+        course_id = get_mandatory_parameter(parameters, "course_id")
+        self.get_course_and_check_rights(course_id)
+        return course_id
+
+    def API_GET(self):
+        bank_courses = [{
+            "id": bank["courseid"],
+            "name": bank["course_name"],
+            "is_removable": self.user_manager.has_admin_rights_on_course(self.course_factory.get_course(bank["courseid"]))
+        } for bank in self.database.problem_banks.find()]
+
+        return 200, bank_courses
+
+    def API_POST(self):
+        course_id = self.get_course_id()
```

Comment:
Anybody can add a course even if he does not have the rights with an API, call. We should check if the course being added has administrative rights of the user making the request

Reply:
That check is was being done in `get_course_id` function. However, I changed that and now is more explicit that rights checking when the course is being gotten.

Reply:
That check is was being done in `get_course_id` function. However, I changed that and now is more explicit that rights checking when the course is being gotten.